lib/internal/modules/esm/initialize_import_meta.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const { getOptionValue } = require('internal/options');
+const experimentalImportMetaResolve =
+getOptionValue('--experimental-import-meta-resolve');
+const { PromisePrototypeThen, PromiseReject } = primordials;
+const asyncESM = require('internal/process/esm_loader');
+
+function createImportMetaResolve(defaultParentUrl) {
+  return async function resolve(specifier, parentUrl = defaultParentUrl) {
+    return PromisePrototypeThen(
+      asyncESM.esmLoader.resolve(specifier, parentUrl),
+      ({ url }) => url,
+      (error) => (
+        error.code === 'ERR_UNSUPPORTED_DIR_IMPORT' ?
+          error.url : PromiseReject(error))
+    );
+  };
+}
+
+function initializeImportMeta(meta, context) {
+  const url = context.url;
+
+  // Alphabetical
+  if (experimentalImportMetaResolve)
+    meta.resolve = createImportMetaResolve(url);
```

Comment:
Any idea what the alphabetical comment is for? (I know you cut this from translators.js and the comment was already there)

Reply:
so at some point in time, there was a bug from people seeing various userland things in differing orders from Object.keys/for...in/etc. so in general if something is a dictionary of sorts it should be alphabetical. So, for example, moving meta.resolve to below meta.url would change the observed key ordering. Keep it alphabetical to prevent github issues.